File: src/main/java/com/github/kory33/signvote/exception/ScoreCountLimitReachedException.java
package com.github.kory33.signvote.exception;

import com.github.kory33.signvote.vote.VotePoint;
import com.github.kory33.signvote.vote.VoteScore;
import lombok.EqualsAndHashCode;
import lombok.Value;
import org.bukkit.entity.Player;

/**
 * Represents an exception thrown when a player attempts to vote with a score
 * whose vote limit has already been reached.
 */
@EqualsAndHashCode(callSuper = false)
@Value
public class ScoreCountLimitReachedException extends Exception {
    Player player;
    VotePoint voteTarget;
    VoteScore voteScore;
}

File: src/main/kotlin/com/github/kory33/signvote/exception/ScoreCountLimitReachedException.kt
package com.github.kory33.signvote.exception

import com.github.kory33.signvote.vote.VotePoint
import com.github.kory33.signvote.vote.VoteScore
import org.bukkit.entity.Player

/**
 * Represents an exception thrown when a player attempts to vote with a score
 * whose vote limit has already been reached.
 */
data class ScoreCountLimitReachedException
constructor(val player: Player, val voteTarget: VotePoint, val voteScore: VoteScore) : Exception()